pallets/stableswap/src/lib.rs

```diff
@@ -578,13 +596,89 @@
 				pool_id,
 				who,
 				shares: share_amount,
-				amounts: vec![AssetBalance { asset_id, amount }],
+				amounts: vec![AssetAmount {
+					asset_id,
+					amount,
+					..Default::default()
+				}],
 				fee,
 			});
 
 			Ok(())
 		}
 
+		/// Remove liquidity from selected pool by specifying exact amount of asset to receive.
+		///
+		/// Parameters:
+		/// - `origin`: liquidity provider
+		/// - `pool_id`: Pool Id
+		/// - `asset_id`: id of asset to receive
+		/// - 'amount': amount of asset to receive
+		/// - 'max_share_amount': Slippage limit. Max amount of shares to burn.
+		///
+		/// Emits `LiquidityRemoved` event when successful.
+		#[pallet::call_index(6)]
+		#[pallet::weight(<T as Config>::WeightInfo::withdraw_asset_amount())]
```

Comment:
would be better if weight would be parametrized based on the number of assets in the pool